integration-tests/tests/analysis-context-bar.js

```diff
@@ -0,0 +1,56 @@
+const _ = require('lodash/fp')
+const { withRegisteredUser, withBilling, withWorkspace, performAnalysisTabSetup } = require('../utils/integration-helpers')
+const {
+  click, clickable, getAnimatedDrawer, findElement, noSpinnersAfter, delay
+} = require('../utils/integration-utils')
+
+
+const testAnalysisContextBarFn = _.flow(
+  withWorkspace,
+  withBilling,
+  withRegisteredUser
+)(async ({ page, token, testUrl, workspaceName }) => {
+  // Navigate to appropriate part of UI (the analysis tab)
+  await performAnalysisTabSetup(page, token, testUrl, workspaceName)
+
+  // Create a runtime
+  // This delay is because the button in this tab is enabled right away, but the page isn't fully loaded
+  // It should not be used commonly, and is only necessary here to support the test in a high-concurrency environment
+  await delay(10000)
```

Comment:
I'm not convinced that it is worth having the 10-second delay in. I ran this test with line 19 commented out 220 times, with concurrency set to 10. The first 100 times I commented out lines 24 to the end, the other times I ran the complete test.

I got 3 failures, only one of which was the issue that the delay is meant to prevent (the other 2 was one around the spinner not appearing after going to the analyses tab, and the other was a server-side error related to Jupyter starting, "Error modifying cloud environment").

I would advocate that we don't introduce the 10-second delay to the test until we see actual failures on PR/integration test runs. @jdcanas said he has never seen this problem without concurrency, so it seems to be related to having high load, and it's not clear that on dev we will experience close to the type of load that we get on our laptops while running the test concurrently.

@kyuksel what are your thoughts?

Reply:
>I would advocate that we don't introduce the 10-second delay to the test until we see actual failures on PR/integration test runs

I'd agree with that..introducing sleep is easy enough if we do see flakiness in CIs

Reply:
I also agree with not including the delay at this time.

Thank you @cahrens and @jdcanas for having been improving the test to reduce the probability of flakiness in the future as flakiness has cost us so much time and pain in the past.

Reply:
works for me